surveys/promise-rejections/survey.md

```diff
@@ -0,0 +1,152 @@
+}).then(() => {
+  throw new Error();  // This will result in a rejection
+}, () => {
+  throw new Error();  // This will result in a rejection
+}).catch(() => {
+  throw new Error();  // This will result in a rejection
+}).finally(() => {
+  throw new Error();  // This will result in a rejection
+});
+
+async function () {
+  throw new Error();  // This will result in a rejection
+}
+
+async function () {
+  if (a === undefined)
+    a();  // This will result in a rejection
+}
+```
+
+## What is an unhandled rejection?
+
+Rejections are designed to be handled asynchronously. In JavaScript, Promises can be handled when awaiting for a promise inside a try/catch block, or via .catch calls:
+
+
+```js
+async function foo() {
+  throw new Error();
+}
+
+foo().catch(() => console.error("an error occured"));  // 1. Handled
+
+try {
+  await foo();  // 2. Handled
+} catch(e) {
+  console.error("an error occured")
+}
+
+foo(); // 3. Unhandled, but execution continues
+
+const rejected = foo(); // 4. Unhandled on current event loop tick
+
+// 4. ... but handled asynchronously when the setTimeout callback is popped from
+// the event loop
+setTimeout(() => rejected.catch(() => console.error("an error occured")), 100);
+```
+
+An unhandled rejection is a rejection which hasn't been handled yet. It might be handled in the future, like example 4, but it might also stay unhandled forever (like example 3).
+
+## Are you currently using Promises, async/await, a mix, or neither?
+
+Assume using as creating your own `new Promise`, `Promise.resolve`, and `async function`; or using libraries that produce Promises (that you use with `.then`/`.catch`, or with `.await`)
+
+(Multiple choice)
+
+  - [ ] Promises
+  - [ ] async/await
+  - [ ] Not writing Promise-based code or using Promise-based libraries
+
+## How are you handling rejections today?
+
+When consuming Promises, async functions or thenables, which of the options below do you use to handle rejections?
+
+(Multiple choice)
+
+  - [ ] `.catch()`
+  - [ ] `try / catch` wrapping an `await` operation
+  - [ ] Ignore the rejection
+  - [ ] Not writing Promise-based code or using Promise-based libraries
+
+## Do you know that Node.js has a global handled for unhandled rejections (`process.on('unhandledRejection')`)? If so, do you use it?
+ 
+  - [ ] I use `process.on('unhandledRejection')`
+  - [ ] I don't use `process.on('unhandledRejection')`
+  - [ ] I didn't know `process.on('unhandledRejection')` existed
+
+## Do you know that Node.js has a flag to change the behavior of unhandled rejections `--unhandled-rejections`? If so, do you use it?
+
+  - [ ] I use `--unhandled-rejections` set to `'strict'`
+  - [ ] I use `--unhandled-rejections` set to `'warn'`
+  - [ ] I use `--unhandled-rejections` set to `'none'`
+  - [ ] I don't use `--unhandled-rejections`
+  - [ ] I didn't know `--unhandled-rejections` existed
+
+## Are you using Promises in customer facing, production applications?
```

Comment:
'customer facing' here is problematic... every line of code has a customer, it just varies who that customer is. I would restructure this to ask something like:

```
Are you using Promises in any of the following kinds of applications? (check all that apply)

[ ] Production-level code
[ ] Tests
[ ] Benchmarks
[ ] Build pipeline / infrastructure
[ ] Examples and Demos
[ ] Other ____
```